recipes/thrift-cpp/build.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/env bash
+
+BOOST_ROOT=$PREFIX
+ZLIB_ROOT=$PREFIX
+LIBEVENT_ROOT=$PREFIX
+
+# If we build in local dev mode, use the bundled OpenSSL
+if [[ "$PRODUCTION" -eq "0" || "$OSTYPE" == "darwin"* ]]; then
+  OPENSSL_ROOT=$PREFIX
+  OPENSSL_ARGS=--with-openssl=$OPENSSL_ROOT
+else
+  OPENSSL_ARGS=
+fi
+
+if [ -d "${PIC_LIB_PATH:-}" ]; then
+  PIC_LIB_OPTIONS="--with-zlib=${PIC_LIB_PATH} "
+fi
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  aclocal -I ./aclocal
+  libtoolize --copy
+  autoconf
+fi
+
+./configure --with-pic --prefix=$PREFIX \
+  --with-c_glib=no \
+  --with-php=no --with-java=no --with-perl=no --with-erlang=no --with-csharp=no \
+  --with-ruby=no --with-haskell=no --with-erlang=no --with-d=no \
+  --with-python=no \
+  --with-boost=${BOOST_ROOT} \
+  --with-zlib=${ZLIB_ROOT} \
+  --with-libevent=${LIBEVENT_ROOT} \
+  --with-nodejs=no \
+  --with-lua=no \
+  --with-go=no --with-qt4=no --with-libevent=no ${PIC_LIB_OPTIONS:-} $OPENSSL_ARGS
+
+make   # Build fails with -j${BUILD_THREADS}
+make install
+cd contrib/fb303
+rm -f config.cache
+chmod 755 ./bootstrap.sh
+./bootstrap.sh --with-boost=${BOOST_ROOT}
+chmod 755 configure
+CPPFLAGS="-I${PREFIX}/include" PY_PREFIX=${PREFIX} ./configure \
+  --with-boost=${BOOST_ROOT} \
+  --with-java=no --with-php=no --prefix=${PREFIX} \
+  --with-thriftpath=${PREFIX} $OPENSSL_ARGS
+make -j"${CPU_COUNT}"
```

Comment:
We don't usually use CPU_COUNT here, because on many CI's, the CPU_COUNT reported is much greater than the actual allocation.  Instead, we are planning on passing MAKEFLAGS through conda-build, and setting MAKEFLAGS in a sane way for each CI platform.